internal/controller/deployment_builder_access.go

```diff
@@ -121,8 +122,87 @@ func (db *DeploymentBuilder) ApplyAccessStrategyToDeployment(
 		return fmt.Errorf("failed to add environment variables to container: %w", err)
 	}
 
-	// Future extensions could be added here:
-	// - sidecar containers
+	// Conditional sidecar logic based on access strategy name
+	if accessStrategy.Name == "aws-ssm-remote-access" {
+		if err := db.addSSMSidecarContainer(deployment, accessStrategy); err != nil {
+			return fmt.Errorf("failed to add SSM sidecar container: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// addSSMSidecarContainer adds an SSM sidecar container to the deployment with shared volume
+func (db *DeploymentBuilder) addSSMSidecarContainer(
+	deployment *appsv1.Deployment,
+	accessStrategy *workspacesv1alpha1.WorkspaceAccessStrategy,
+) error {
+	// Create shared volume for communication between main container and sidecar
+	sharedVolume := corev1.Volume{
+		Name: "ssm-remote-access",
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{
+				SizeLimit: func() *resource.Quantity {
+					limit := resource.MustParse("1Gi")
+					return &limit
+				}(),
+			},
+		},
+	}
+
+	// Create volume mount for shared directory
+	sharedVolumeMount := corev1.VolumeMount{
+		Name:      "ssm-remote-access",
+		MountPath: "/ssm-remote-access",
+	}
+
+	// Get sidecar image from access strategy controller configuration
+	var sidecarImage string
+	if accessStrategy.Spec.ControllerConfig != nil {
+		sidecarImage = accessStrategy.Spec.ControllerConfig["SSM_SIDECAR_IMAGE"]
+	}
+
+	if sidecarImage == "" {
+		return fmt.Errorf("SSM_SIDECAR_IMAGE environment variable not found in access strategy %s", accessStrategy.Name)
+	}
+
+	ssmContainer := corev1.Container{
+		Name:    "ssm-agent-sidecar",
+		Image:   sidecarImage,
+		Command: []string{"/bin/sh"},
+		Args: []string{
+			"-c",
+			"cp /usr/local/bin/remote-access-server /ssm-remote-access/ || echo \"Failed to copy: $?\"; sleep infinity",
```

Comment:
this make it hard to reuse for other use case. Can we add a script in the side car image? The script can be enhance to support more use cases.